commons/src/main/scala/org/bitbucket/inkytonik/cooma/SemanticAnalyser.scala

```diff
@@ -26,13 +26,15 @@ class SemanticAnalyser(
     import org.bitbucket.inkytonik.kiama.util.Messaging.{check, collectMessages, error, Messages, noMessages}
     import org.bitbucket.inkytonik.cooma.CoomaParserPrettyPrinter.show
     import org.bitbucket.inkytonik.cooma.CoomaParserSyntax._
-    import org.bitbucket.inkytonik.cooma.SymbolTable.primitivesTypesTable
 
     val decorators = new Decorators(tree)
     import decorators._
 
     lazy val errors : Messages =
         collectMessages(tree) {
+            case i @ IdnDef(s) if isPrimitiveTypeName(s) =>
+                error(i, s"$s is a reserved primitive type name")
+
```

Comment:
```
cooma> val x = { String = 1 }
x : { String : Int } = { String = 1 }
```

I suppose this is okay, but we will need to be careful if we introduce imports or destructors.

Reply:
Yeah, allowing user identifiers be the same as pre-defined names is a tricky one. We've now moved to allowing over-riding of things like `Int` and `String` instead of treating them as keywords as many languages do. In the end I think it's best to keep the abstract syntax as simple as possible.